pyautoenv.py

```diff
@@ -203,29 +208,32 @@ def has_poetry_env(directory: str) -> bool:
     return os.path.isfile(os.path.join(directory, "poetry.lock"))
 
 
-def poetry_activator(args: Args) -> str | None:
+def poetry_activator(args: Args) -> Union[str, None]:
     """
     Return the activator for the venv associated with a poetry project directory.
 
     If there are multiple poetry environments, pick the one with the
     latest modification time.
     """
-    if env_list := poetry_env_list(args.directory):
+    env_list = poetry_env_list(args.directory)
+    if env_list:
         env_dir = max(env_list, key=lambda p: os.stat(p).st_mtime)
         return activator(env_dir, args)
     return None
 
 
-def poetry_env_list(directory: str) -> list[str]:
+def poetry_env_list(directory: str) -> List[str]:
     """
     Return list of poetry environments for the given directory.
 
     This can be found via the poetry CLI using
     ``poetry env list --full-path``, but it's painfully slow.
     """
-    if (cache_dir := poetry_cache_dir()) is None:
+    cache_dir = poetry_cache_dir()
+    if cache_dir is None:
         return []
-    if (env_name := poetry_env_name(directory)) is None:
+    env_name = poetry_env_name(directory)
+    if (env_name) is None:
```

Comment:
Just missed these braces here, then I think this is good to go 👍 

```suggestion
    if env_name is None:
```

Reply:
It's fixed